src/platform/fs-provider.ts

```diff
@@ -51,6 +51,11 @@ export class PlatformFS implements vscode.FileSystemProvider {
     content: Uint8Array,
     options: { create: boolean; overwrite: boolean }
   ): Promise<void> {
+    const apiId = getApiId(uri)!;
```

Comment:
I think we should throw NoPermissions error here, instead of a generic error: https://code.visualstudio.com/api/references/vscode-api#FileSystemError.NoPermissions

Reply:
Done